### ISSUE TITLE
Error in watch mode

### ISSUE THREAD

**Issue by arvinxx** [shelfio/jest-mongodb#1]
When I use the config file in watch mode. It report error below:
```
Error: Database instance is not running. You should start database by calling start() method. BTW it should start automatically if opts.autoStart!=false. Also you may provide opts.debug=true for more info.
    at MongoMemoryServer._callee6$ (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:304:23)
    at tryCatch (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:114:21)
    at step (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:191)
    at /Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:437
    at new Promise (<anonymous>)
    at MongoMemoryServer.<anonymous> (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:99)
    at MongoMemoryServer.getInstanceData (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:315:22)
    at MongoMemoryServer._callee5$ (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:251:29)
    at tryCatch (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/arvin/Desktop/koa-app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:114:21)
    at step (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:191)
    at /Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:437
    at new Promise (<anonymous>)
    at MongoMemoryServer.<anonymous> (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:79:99)
    at MongoMemoryServer.stop (/Users/arvin/Desktop/koa-app/node_modules/mongodb-memory-server/lib/MongoMemoryServer.js:283:22)
    **at module.exports (/Users/arvin/Desktop/koa-app/test/config/teardown.js:3:27)**
    at /Users/arvin/Desktop/koa-app/node_modules/jest-cli/build/run_jest.js:242:49
    at Generator.next (<anonymous>)
    at step (/Users/arvin/Desktop/koa-app/node_modules/jest-cli/build/run_jest.js:81:191)
```
In error message I find may `teardown.js` file go wrong.And after I comment line 3.
```js
   await global.__MONGOD__.stop();
```
It works well.
So maybe this code should be refactored to suit watch mode?
Now I use it with that line commented.

**Comment by vladgolubev** [comment]
Hey @arvinxx 

thanks for bringing thi sup. This is tricky. I never run mongodb tests in watch mode, as it does lots of heavy operations and can greatly degrade machine performance. It does heavy stuff with mongo, so it's quite unbearable with ~500 tests.

You solution may work for you, but then `mongod` won't always stop when tests are finished.

I see the only hacky solution - wrap this code into `process.on('exit')` - but need to try does it work properly.

Thoughts?

**Comment by azangru** [comment]
@vladgolubev @arvinxx Are there any downsides to doing all the setup and teardown logic only in the file that is used for defining the test environment (`mongo-environment.js` in this repo)? As is described by someone in [this blog post](https://itnext.io/parallel-testing-a-graphql-server-with-jest-44e206f3e7d2)? This means that `mongodb` will be set up and torn down for every test suite (not once before all test suites), so it should work with watch mode without any problems. I am just a bit afraid that this might not scale.

**Comment by vladgolubev** [comment]
@azangru @arvinxx  I just updated `jest` and `mongodb` to the latest versions + removed `--runInBand` flag. This should work much more reliably now ✨  Would you mind giving it a spin now?

**Comment by RodolfoSilva** [comment]
Fixed by #57

**Comment by RodolfoSilva** [comment]
@vladgolubev fixed by #57
